app/controllers/application_controller.rb

```diff
@@ -749,7 +749,9 @@ def schedule_enable_disable(schedules, enabled)
   # Build the user_emails hash for edit screens needing the edit_email view
   def build_user_emails_for_edit
     @edit[:user_emails] = {}
+    group_id = User.find_by_userid(session[:userid])[:current_group_id]
     User.all.sort_by { |u| u.name.downcase }.each do |u|
+      next unless u.current_group_id == group_id
```

Comment:
I think this could have unwanted behavior in environments where users are members of multiple groups. The list would end up being different each time a user switches his current group using the pull-down in the upper right corner of the UI.

It would be better to build the list with the users that have membership in all the groups that the current user is a member. I'm not sure it that would still meet you requirements for restricting the list.

I think this query would do that but you'd have to write a test to verify that.
```
user = User.find_by_userid(session[:userid])[:current_group_id]
User.includes(:miq_groups).where(:miq_groups => {:id => user.miq_groups.map(&:id)}).uniq.sort_by {...
```